expat/tests/common.c

```diff
@@ -191,18 +192,18 @@ enum XML_Status
 _XML_Parse_SINGLE_BYTES(XML_Parser parser, const char *s, int len,
                         int isFinal) {
   const int chunksize = g_chunkSize;
-  int offset = 0;
   if (chunksize > 0) {
     // parse in chunks of `chunksize` bytes as long as possible
-    for (; offset + chunksize < len; offset += chunksize) {
-      enum XML_Status res = XML_Parse(parser, s + offset, chunksize, XML_FALSE);
+    for (; chunksize <= len; len -= chunksize, s += chunksize) {
```

Comment:
Sorry, I missed this previously:

Why `chunksize <= len`? It means the `XML_Parse()` at line 206 will have len=0 if the `len` arg is divisible by `chunksize`, essentially adding an extra `XML_Parse()` call that was not previously there.

Note that for chunksize 1, len will always be divisible, so we'll never make a `len=1 isFinal=1` call, just some amount of `len=1 isFinal=0` and the concluding `len=0 isFinal=1`.

It probably doesn't matter in practice for now, but could potentially help hide problems in the future.

Reply:
@Snild-Sony nice catches, maybe that's the last drop needed to reveal this is the wrong approach altogether. Let's flip this around and make `_XML_Parse_SINGLE_BYTES` reject pathological input like `s == NULL` loudly so that the tests in `test_empty_parse` have to resort to plain `XML_Parse` and keep testing the raw actual `XML_Parse` API. Pushing in a second…

Reply:
I just wanted it to be `chunksize < len`  instead, then it would've been perfect... :sob:

Reply:
@Snild-Sony I have re-added a refactoring of `_XML_Parse_SINGLE_BYTES` using `chunksize < len` as a dedicated commit just for you now :smiley: The assertion still blocks pathological input altogether.

Reply:
@Snild-Sony PS: I should note that GitHub web UI is behind in displayed pushed changes at the moment. Branch https://github.com/libexpat/libexpat/commits/tests-more-chunk-size-coverage has the commit but the pull request does not show it :scream: PPS: As with all status pages, https://www.githubstatus.com/ is all green, lovely.